lib/figaro.rb

```diff
@@ -0,0 +1,63 @@
+require 'yaml'
+require 'erb'
+
+class Figaro
+  def self.env
+    @env ||= Environment.new
+  end
+
+  def self.setup(path)
+    env.setup(path)
+  end
+
+  def self.require_keys(keys)
+    env.require_keys(keys)
+  end
+
+  class Environment
+    def setup(path, env = Rails.env)
+      @config = {}
+      values = YAML.safe_load(File.read(path))
+
+      keys = Set.new(values.keys - %w[production development test])
+      keys |= values[env].keys
+
+      keys.each do |key|
+        value = values[env][key] || values[key]
+
+        ENV[key] = value if key == key.upcase
+
+        @config[key] = value
+      end
+
+      @config
+    end
+
+    def respond_to_missing?(method_name, _include_private = false)
+      key = method_name.to_s.tr('?!', '')
+      @config.key?(key)
+    end
+
+    def method_missing(method, *_args)
+      string_key = method.to_s
+      key = string_key.tr('?!', '')
+      raise_exception = string_key.ends_with?('!')
```

Comment:
I'm not able to find where this behavior of raising an exception in response to a missing key ending in `!` is coming from. Is it a Figaro behavior?

Reply:
Yep, Figaro would raise if called with a `!` and the key didn't exist. It would return true/false if the method called ended with `?`.